core/state_transition.go

```diff
@@ -235,30 +238,42 @@ func (st *StateTransition) TransitionDb() (ret []byte, usedGas uint64, failed bo
 	if executionMgrTime.Cmp(big.NewInt(0)) == 0 {
 		executionMgrTime = big.NewInt(1)
 	}
+
+	// TODO: queue origin is always 0 with current version of em
+	queueOrigin := big.NewInt(0)
```

Comment:
Oh amazing so just double checked & getting the L1MsgSender does not make any checks wrt what queue we're on -- https://github.com/ethereum-optimism/optimism-monorepo/blob/master/packages/contracts/contracts/optimistic-ethereum/ovm/ExecutionManager.sol#L1078

So it's totally safe in the contracts to just always keep the queue origin zero